pyjobshop/solvers/ortools/Objective.py

```diff
@@ -29,7 +29,19 @@ def _makespan_expr(self) -> LinearExprT:
         Returns an expression representing the makespan of the model.
         """
         makespan = self._model.new_int_var(0, MAX_VALUE, "makespan")
-        completion_times = [var.end for var in self._task_vars]
+        completion_times = []
+
+        for task, var in zip(self._data.tasks, self._task_vars):
+            if task.optional:
+                # When the task is absent, it should not restrict the makespan.
```

Comment:
Need to document this better. This is only when the task end variable is bounded. It's normally free when it's absent but there are edge cases where it's not (e.g., when earliest_end is set).